crux_core/src/capability.rs

```diff
@@ -1,14 +1,231 @@
+//! Capabilities provide a user-friendly API to request side-effects from the shell.
```

Comment:
We might want to put `#![warn(missing_docs)]` on this at some point?  Makes it harder to forget to add them.  Ofc it's also annoying af when you're working on stuff, so maybe one for later